<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Making Venn and Euler Diagrams in R</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 60px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 65px;
  margin-top: -65px;
}

.section h2 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h3 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h4 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h5 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h6 {
  padding-top: 65px;
  margin-top: -65px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Kristin Ashby</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="vennDiagram_v5.html">Posts</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="http://github.com/kmashby">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://www.linkedin.com/in/kristin-mceuen-ashby-94573685/">
    <span class="fa fa-linkedin fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Making Venn and Euler Diagrams in R</h1>

</div>


<hr />
<div id="a-quick-tutorial-focusing-on-the-different-input-formats-and-features-of-four-venn-diagram-packages." class="section level5">
<h5>A quick tutorial focusing on the different input formats and features of four Venn diagram packages.</h5>
<p></br> <strong>Not interested in using r? <a href="#NonProgramming">Click here</a> to find links to non-programming, web-based solutions.</strong> </br> </br></p>
</div>
<div id="summary-of-packages-in-tutorial" class="section level4">
<h4>Summary of packages in tutorial:</h4>
<ul>
<li><a href="#Venn"><strong>Venn</strong></a>
<ul>
<li>Best use: <strong>Fast venn diagrams</strong></li>
<li>Can make a venn diagram with up to <strong>7 sets</strong>.</li>
</ul></li>
<li><a href="#VennDiagram"><strong>VennDiagram</strong></a>
<ul>
<li>Best use: <strong>Publication worthy Venn diagrams</strong></li>
<li>Makes venn diagrams with up to <strong>4 sets</strong> or euler diagrams with up to 3 sets.</li>
<li>High resolution, options for customization</li>
</ul></li>
<li><a href="#Eulerr"><strong>Eulerr</strong></a>
<ul>
<li>Best use: <strong>Euler diagrams</strong></li>
<li>Great default settings for making beautiful Euler diagrams</li>
<li>Watch out for missing relationships… Euler diagrams can be tricky sometimes.</li>
</ul></li>
<li><a href="#Limma"><strong>Limma</strong></a>
<ul>
<li>Best use: <strong>Venn diagrams after microarray analysis</strong></li>
<li>Can make a venn diagram with up to <strong>5 sets</strong>.</li>
<li>Input is a count matrix rather than a list of IDs.</li>
</ul></li>
</ul>
<p></br></p>
</div>
<div id="create-sample-dataset" class="section level3">
<h3>Create Sample Dataset</h3>
<div id="this-sample-dataset-consists-of-three-samples-and-one-control.-all-have-the-same-genes-ids-but-different-p-values-pval." class="section level5">
<h5>This sample dataset consists of three samples and one control. All have the same genes (IDs) but different p-values (pval).</h5>
</div>
<div id="we-want-to-find-out-how-many-genes-are-significant-in-each-sample-and-how-many-of-those-genes-overlap-with-the-other-samples." class="section level5">
<h5>We want to find out how many genes are significant in each sample and how many of those genes overlap with the other samples.</h5>
<pre class="r"><code># Sample dataset ####
set.seed(23)

IDs &lt;- paste(&quot;Gene&quot;, 1:100, sep = &quot;_&quot;)

sample1 &lt;- data.frame(IDs, pval = rnorm(length(IDs)))
sample2 &lt;- data.frame(IDs, pval = rnorm(length(IDs)))
sample3 &lt;- data.frame(IDs, pval = rnorm(length(IDs)))
control &lt;- data.frame(IDs, pval = runif(length(IDs)), 0.01, 1)

# Get significant genes in each sample.
sig1 = sample1[sample1$pval &lt; 0.05,]
sig2 = sample2[sample2$pval &lt; 0.05,]
sig3 = sample3[sample3$pval &lt; 0.05,]
sigcntl = control[control$pval &lt; 0.05,]</code></pre>
</div>
</div>
<div id="Venn" class="section level3">
<h3>Venn</h3>
<p><a href="https://cran.r-project.org/web/packages/venn/venn.pdf" class="uri">https://cran.r-project.org/web/packages/venn/venn.pdf</a></p>
<div id="venn-can-create-a-venn-diagram-with-up-to-7-sets-and-is-quick-and-easy-to-use." class="section level5">
<h5>Venn can create a venn diagram with up to 7 sets and is quick and easy to use.</h5>
<pre class="r"><code>library(venn)

# Input is list of IDs

# Get all significant genes (p values &lt; 0.05)
ID_list &lt;- list(Sample1=sample1$IDs[sample1$pval &lt; 0.05], 
                Sample2=sample2$IDs[sample2$pval &lt; 0.05], 
                Sample3=sample3$IDs[sample3$pval &lt; 0.05], 
                Control=control$IDs[control$pval &lt; 0.05])

venn(ID_list,
     cexil = 1.5, # Change intersection labels
     cexsn = 1.25, # Change set name labels
     zcolor = c(&quot;blue&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;green&quot;), opacity = 0.5)</code></pre>
<p><img src="vennDiagram_v5_files/figure-html/unnamed-chunk-3-1.png" width="384" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="VennDiagram" class="section level3">
<h3>VennDiagram</h3>
<p><a href="https://cran.r-project.org/web/packages/VennDiagram/VennDiagram.pdf" class="uri">https://cran.r-project.org/web/packages/VennDiagram/VennDiagram.pdf</a></p>
<p>VennDiagram has many options for cutsomization and can be used to create high quality figures.</p>
<p>You can use it to make both Venn and Euler diagrams.</p>
<pre class="r"><code>library(VennDiagram)

# Input is list of IDs
ID_list &lt;- list(Sample1=sig1$IDs, Sample2=sig2$IDs, Sample3=sig3$IDs, Control=sigcntl$IDs)

myVenn &lt;- venn.diagram(x=ID_list[1:4], filename = NULL,
                     fill = c(&quot;blue&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;green&quot;), alpha = 0.5, col=&quot;transparent&quot;,
                     cex = 1.25, fontface=&quot;bold&quot;, # Intersection label size and style
                     cat.fontface=&quot;bold&quot;, cat.cex = 1, # Category name size and style
                     cat.col = c(&quot;navy&quot;, &quot;darkred&quot;, &quot;goldenrod&quot;, &quot;darkgreen&quot;))
   
grid.newpage()
grid.draw(myVenn)</code></pre>
<p><img src="vennDiagram_v5_files/figure-html/unnamed-chunk-4-1.png" width="384" style="display: block; margin: auto;" /></p>
<pre class="r"><code>myVenn &lt;- venn.diagram(x=ID_list[1:3], filename = NULL, fill = c(&quot;blue&quot;, &quot;red&quot;, &quot;yellow&quot;), 
                     alpha = 0.5, col=&quot;transparent&quot;,
                     cex = 1.5, fontface=&quot;bold&quot;, # Intersection label size and style
                     cat.cex=1.5, cat.fontface=&quot;bold&quot;, # Category name size and style 
                     cat.dist=c(-0.01,-0.01,-0.01), # Category name distance from edge of the circle 
                     cat.pos = c(325,35,180), # Category name position in degrees
                     cat.col = c(&quot;navy&quot;, &quot;darkred&quot;, &quot;goldenrod&quot;),
                     label.col = c(&quot;navy&quot;, &quot;white&quot;, &quot;darkred&quot;, &quot;white&quot;, &quot;black&quot;, &quot;white&quot;, &quot;goldenrod&quot;))
   
grid.newpage()
grid.draw(myVenn)</code></pre>
<p><img src="vennDiagram_v5_files/figure-html/unnamed-chunk-4-2.png" width="384" style="display: block; margin: auto;" /></p>
<pre class="r"><code>myVenn &lt;- venn.diagram(x=ID_list[c(1,3,4)], fill = c(&quot;blue&quot;, &quot;red&quot;, &quot;yellow&quot;),
                     alpha = 0.5, fontface = &quot;bold&quot;,
                     cat.fontface = &quot;bold&quot;,  
                     lty = &quot;dashed&quot;, lwd=0.75,
                     filename = NULL)

grid.newpage()
grid.draw(myVenn)</code></pre>
<p><img src="vennDiagram_v5_files/figure-html/unnamed-chunk-4-3.png" width="384" style="display: block; margin: auto;" /></p>
<pre class="r"><code>myVenn &lt;- venn.diagram(x=ID_list[c(1,4)], fill = c(&quot;blue&quot;, &quot;yellow&quot;),
                     alpha = 0.5, fontface = &quot;bold&quot;,
                     cat.fontface = &quot;bold&quot;, 
                     cat.dist = c(-0.1,0.03), cat.pos = c(-30,25),
                     lty = &quot;dashed&quot;, lwd=0.75,
                     filename = NULL)

grid.newpage()
grid.draw(myVenn)</code></pre>
<p><img src="vennDiagram_v5_files/figure-html/unnamed-chunk-4-4.png" width="384" style="display: block; margin: auto;" /></p>
<hr />
</div>
<div id="Countmatrix" class="section level3">
<h3>Input data: Creating set relationship or count matrix</h3>
<div id="both-limma-and-eulerr-do-not-accept-a-list-of-ids-as-input.-you-can-easily-change-a-list-of-ids-into-a-count-matrix.-here-is-one-way-to-do-that" class="section level5">
<h5>Both Limma and Eulerr do not accept a list of IDs as input. You can easily change a list of IDs into a count matrix. Here is one way to do that:</h5>
<pre class="r"><code>library(reshape2)

ID_list_m &lt;- melt(ID_list[1:length(ID_list)])
ID_counts &lt;- dcast(ID_list_m, value ~ L1, fun.aggregate = length)

head(ID_counts)</code></pre>
<pre><code>##      value Control Sample1 Sample2 Sample3
## 1  Gene_10       0       0       1       1
## 2 Gene_100       0       0       1       0
## 3  Gene_11       0       0       1       1
## 4  Gene_12       0       1       0       0
## 5  Gene_13       0       1       0       1
## 6  Gene_14       0       0       0       1</code></pre>
</div>
</div>
<div id="Eulerr" class="section level3">
<h3>Eulerr</h3>
<p>Eulerr’s default settings for creating Euler diagrams are really beautiful. Input data is a <a href="#Countmatrix">count matrix</a>.</p>
<p><a href="https://cran.r-project.org/web/packages/eulerr/vignettes/introduction.pdf" class="uri">https://cran.r-project.org/web/packages/eulerr/vignettes/introduction.pdf</a></p>
<pre class="r"><code>library(eulerr)

# Input is a count matrix
fit &lt;- euler(ID_counts[2:4])
plot(fit, quantities=TRUE)</code></pre>
<p><img src="vennDiagram_v5_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<div id="however-eulerr-diagrams-can-be-tricky-sometimes" class="section level5">
<h5>However, Eulerr diagrams can be tricky sometimes …</h5>
<p>Notice that Eulerr is missing the overlap between Sample 2 and the Control (Gene_39):</p>
<table>
<thead>
<tr class="header">
<th align="left">value</th>
<th align="right">Control</th>
<th align="right">Sample1</th>
<th align="right">Sample2</th>
<th align="right">Sample3</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Gene_27</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">Gene_30</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">Gene_39</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">Gene_65</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
</div>
<div id="gene_39-is-also-missing-below.-some-relationships-are-difficult-to-show-in-euler-diagrams." class="section level5">
<h5>Gene_39 is also missing below. Some relationships are difficult to show in euler diagrams.</h5>
<pre class="r"><code>fit &lt;- euler(ID_counts[2:5])
plot(fit, quantities=TRUE)</code></pre>
<p><img src="vennDiagram_v5_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
</div>
</div>
<div id="Limma" class="section level3">
<h3>Limma</h3>
<p><a href="https://bioconductor.org/packages/devel/bioc/manuals/limma/man/limma.pdf" class="uri">https://bioconductor.org/packages/devel/bioc/manuals/limma/man/limma.pdf</a></p>
<p>Limma’s vennDiagram function can plot up to five sets and accepts a matrix of counts rather than a list of IDs. <a href="#Countmatrix">Click here</a> to see sample data.</p>
<p>Count matrix:</p>
<pre><code>##      value Control Sample1 Sample2 Sample3
## 1  Gene_10       0       0       1       1
## 2 Gene_100       0       0       1       0
## 3  Gene_11       0       0       1       1
## 4  Gene_12       0       1       0       0
## 5  Gene_13       0       1       0       1
## 6  Gene_14       0       0       0       1</code></pre>
<p>Notice that it also shows the genes that were not signicant in any of the samples (bottom right corner).</p>
<pre class="r"><code>#biocLite(&quot;limma&quot;)
library(&quot;limma&quot;)

vennDiagram(ID_counts[2:4], include = &quot;both&quot;, 
            names = c(&quot;Sample 1&quot;, &quot;Sample 2&quot;, &quot;Sample 3&quot;), 
            circle.col=c(&quot;blue&quot;, &quot;red&quot;, &quot;yellow&quot;))</code></pre>
<p><img src="vennDiagram_v5_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
<pre class="r"><code>vennDiagram(ID_counts[2:5], include = &quot;both&quot;, 
            names = c(&quot;Sample 1&quot;, &quot;Sample 2&quot;, &quot;Sample 3&quot;, &quot;Control&quot;), 
            circle.col=c(&quot;blue&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;green&quot;))</code></pre>
<p><img src="vennDiagram_v5_files/figure-html/unnamed-chunk-10-2.png" width="672" /></p>
<p>Limma was written for processing microarray data and the venn diagram function works well following this pipeline:</p>
<pre class="r"><code>fit &lt;- eBayes(lmFit(Y,x))
results &lt;- decideTests(fit)
a &lt;- vennCounts(results)
vennDiagram(a)</code></pre>
<p>See <a href="https://rdrr.io/bioc/limma/man/ebayes.html" class="uri">https://rdrr.io/bioc/limma/man/ebayes.html</a> for more information</p>
<hr />
</div>
<div id="NonProgramming" class="section level1">
<h1>Non programming solutions</h1>
<p>If you just want to make a quick venn diagram, here are some online tools to help you:</p>
<ul>
<li>Venny
<ul>
<li><a href="http://www.forestrylab.org/tool/Venny/" class="uri">http://www.forestrylab.org/tool/Venny/</a></li>
<li><a href="http://bioinfogp.cnb.csic.es/tools/venny/" class="uri">http://bioinfogp.cnb.csic.es/tools/venny/</a></li>
<li>Up to 4 different sets</li>
<li>Not proportional</li>
<li>Can include percents or counts</li>
</ul></li>
<li>BioVenn
<ul>
<li><a href="http://www.biovenn.nl/" class="uri">http://www.biovenn.nl/</a></li>
<li>Up to 3 different sets</li>
<li>Proportional</li>
</ul></li>
<li>InteractiVenn
<ul>
<li><a href="http://www.interactivenn.net/" class="uri">http://www.interactivenn.net/</a></li>
<li>Up to 6 different sets</li>
<li>Not proportional</li>
</ul></li>
<li>Venn by Bioinformatics and Evolutionary Genomics
<ul>
<li><a href="http://bioinformatics.psb.ugent.be/webtools/Venn/" class="uri">http://bioinformatics.psb.ugent.be/webtools/Venn/</a></li>
<li>Up to 3 different sets</li>
</ul></li>
</ul>
<p></br></p>
<p></br></p>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
